director.go

```diff
@@ -0,0 +1,200 @@
+package stashcp
+
+import (
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type DirectorCache struct {
+	ResourceName string
+	EndpointUrl  string
+	Priority     int
+	AuthedReq    bool
+}
+
+// Simple parser to that takes a "values" string from a header and turns it
+// into a map of key/value pairs
+func HeaderParser(values string) (retMap map[string]string) {
+	retMap = map[string]string{}
+
+	// Some headers might not have values, such as the
+	// X-OSDF-Authorization header when the resource is public
+	if values == "" {
+		return
+	}
+
+	mapPairs := strings.Split(values, ",")
+	for _, pair := range mapPairs {
+		// Remove any unwanted spaces
+		pair = strings.ReplaceAll(pair, " ", "")
+
+		// Break out key/value pairs and put in the map
+		split := strings.Split(pair, "=")
+		retMap[split[0]] = split[1]
+	}
+
+	return retMap
+}
+
+// Given the Director response, create the ordered list of caches
+// and store it as namespace.SortedDirectorCaches
+func CreateNSFromDirectorResp(dirResp *http.Response, namespace *Namespace) (err error) {
```

Comment:
I think this is fine, but I see the letters "NSF" in here like this, and I think of governmental agencies.